src/main/java/com/gard/testbed/activitySystem/IState.java

```diff
@@ -3,6 +3,19 @@
 import java.util.List;
 
 public interface IState {
+
+    /**
+     * Returns true if all outgoing transition triggers test true against blackboard
+     * @return
+     */
+    boolean isCompleted(IBlackboard player);
```

Comment:
You don't need all outgoing, you need just one of them. I'd more then one you should choose one. (Assuming we are not going concurrent for now)